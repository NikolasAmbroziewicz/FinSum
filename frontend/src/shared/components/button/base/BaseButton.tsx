import { ButtonTheme } from "./types";

export interface IBaseButton {
  handler?: () => void;
  children: JSX.Element | string;
  type?: 'button' | 'submit' | 'reset' | undefined;
  color?: ButtonTheme,
  styles?: string
}

const BaseButton: React.FC<IBaseButton> = ({ color = ButtonTheme.base, handler, type, styles, children }) => {

  const buttonColor = () => {
    return color === ButtonTheme.base ? 'bg-sky-500 hover:bg-sky-600' : 'bg-red-600 hover:bg-red-500'
  }

  return (
    <button
      onClick={handler}
      className={`${buttonColor()} text-white py-2 px-4 rounded ${styles}`}
      type={type}
    >
      {children}
    </button>
  );
};

export default BaseButton;
