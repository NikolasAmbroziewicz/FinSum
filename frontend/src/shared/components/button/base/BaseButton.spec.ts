import { describe, expect, it } from 'vitest';
import BaseButton, { IBaseButton } from './BaseButton';

const baseButtonProps: IBaseButton = {
  children: 'Submit',
  type: 'button'
};

describe('BaseButton', () => {
  it('defined', () => {
    expect(BaseButton(baseButtonProps)).toMatchSnapshot();
  });
});
