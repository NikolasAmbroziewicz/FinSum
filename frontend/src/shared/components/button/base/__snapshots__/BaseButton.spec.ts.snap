// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseButton > defined 1`] = `
<button
  className="bg-sky-500 hover:bg-sky-600 text-white py-2 px-4 rounded "
  type="button"
>
  Submit
</button>
`;
